creme/utils/math.py

```diff
@@ -38,7 +38,9 @@ def dot(x: dict, y: dict):
         42
 
     """
-    return sum(xi * y.get(i, 0) for i, xi in min(x, y, key=len).items())
+    l = min(x, y, key=len)
+    r = y if l is x else x
```

Comment:
I don't really like `if l is x` because it has to compare `dict`s. Can you instead do this:

```python
if len(l) < len(r):
    l, r = x, y
else:
    l, r = y, x
```

Or something like that?